### PR TITLE
docs: Update Google SMTP Setup to Use Application-Specific Passwords

### LINKED ISSUE #1025: Documentation update: Gmail SMTP settings
Hi folks
Just want to suggest an update to the documentation here: 
https://github.com/overhangio/tutor/blob/master/docs/tutorials/google-smtp.rst

specifically related to the "less secure apps" part, this feature is depreciated by Google, users should instead refer to the Application-Specific Passwords feature (https://support.google.com/mail/answer/185833?hl=en) to allow less secure apps to have access to Gmail

Refer below forum post for details:
https://discuss.openedx.org/t/issues-regarding-smtp/12594

### PR DESCRIPTION
**Pull request by Abdul-Muqadim-Arbisoft**
Resolved #1025 
This pull request updates the documentation for setting up Google Mail as an SMTP server in Tutor. It replaces the deprecated "Less Secure Apps" access method with "Application-Specific Passwords" as recommended by Google for enhanced security.

Key Changes:
- Removed instructions related to enabling "Less Secure App Access," which Google has deprecated.
- Added detailed steps to set up "Application-Specific Passwords," including enabling 2-Step Verification and generating a secure 16-digit password.
- Included links to Google's official support page for creating app-specific passwords and a relevant discussion from the Open edX community forum that highlights issues related to SMTP setup.

### CODE REVIEW

File: docs/tutorials/google-smtp.rst [hunk 1]
```diff
@@ -6,7 +6,18 @@ By default, Tutor comes with a simple SMTP server for sending emails. Such a ser
 .. warning::
   Google Mail SMTP servers come with their own set of limitations. For instance, you are limited to sending 500 emails a day. Reference: https://support.google.com/mail/answer/22839
 
-You should authorize third-party to access your Google Mail account. In your Google Mail account, select "Manage Account", "Security", and turn on "Less Secure App Access". Check the Google documentation for more information on "less secure apps": https://support.google.com/accounts/answer/6010255
+Authorization for Third-Party Access :
+
+Google has deprecated the "Less Secure App Access." Instead, it is recommended to use "Application-Specific Passwords" for more secure access:
```
[the remark this answers is not in the document]
Comment (Abdul-Muqadim-Arbisoft) [reply docs/tutorials/google-smtp.rst]: concern addressed

File: docs/tutorials/google-smtp.rst [hunk 2]
```diff
@@ -6,7 +6,18 @@ By default, Tutor comes with a simple SMTP server for sending emails. Such a ser
...
+
+1. Ensure 2-Step Verification is enabled on your Google Account.
+2. Visit Google Account Security.
+3. Under "Signing in to Google," choose "App passwords."
+4. You might need to sign in again. Once you do, select "Select app" and choose "Other (Custom name)" from the dropdown.
+5. Enter a name that helps you remember the purpose of this password, like "Tutor SMTP".
+6. Generate and note your 16-character app-specific password.
```
[the remark this answers is not in the document]
Comment (Abdul-Muqadim-Arbisoft) [reply docs/tutorials/google-smtp.rst]: concern addressed

File: docs/tutorials/google-smtp.rst [hunk 3]
```diff
@@ -6,7 +6,18 @@ By default, Tutor comes with a simple SMTP server for sending emails. Such a ser
 .. warning::
   Google Mail SMTP servers come with their own set of limitations. For instance, you are limited to sending 500 emails a day. Reference: https://support.google.com/mail/answer/22839
 
-You should authorize third-party to access your Google Mail account. In your Google Mail account, select "Manage Account", "Security", and turn on "Less Secure App Access". Check the Google documentation for more information on "less secure apps": https://support.google.com/accounts/answer/6010255
+Authorization for Third-Party Access :
+
+To enhance security, Google recommends the use of "Application-Specific Passwords" for third-party access to Google services:
+
```
Comment (DawoudSheraz) [inline docs/tutorials/google-smtp.rst:12]: After `Google recommends the use of "Application-Specific Passwords" for third-party access to Google services.`, let's add a small sentence to emphasize on following the steps to enable the feature.

File: docs/tutorials/google-smtp.rst [hunk 4]
```diff
@@ -6,7 +6,18 @@ By default, Tutor comes with a simple SMTP server for sending emails. Such a ser
...
+1. Activation of 2-Step Verification is required for the Google Account.
```
Comment (DawoudSheraz) [inline docs/tutorials/google-smtp.rst:13]: nit: Activate 2-step verification for Google Account.

Comment (Abdul-Muqadim-Arbisoft) [reply docs/tutorials/google-smtp.rst]: Addressed

File: docs/tutorials/google-smtp.rst (hunk 3 above)
Comment (Abdul-Muqadim-Arbisoft) [reply docs/tutorials/google-smtp.rst]: concern addressed

**Pull request merged**